psydac/utilities/utils.py

```diff
@@ -5,13 +5,32 @@
 import numpy as np
 
 #===============================================================================
-def refine_array_1d( x, n ):
+def refine_array_1d(x, n, remove_duplicates=True):
+    """Refines a 1D array by create a subdivision between each point
 
-    xr = []
-    for (a,b) in zip(x[:-1],x[1:]):
-        xr.extend( np.linspace( a, b, n, endpoint=False ) )
-    xr.append( x[-1] )
+    Parameters
+    ----------
+    x : ndarray
+        1D array to be refined
+    n : int
+        The number of points in each subdivision
+
+    remove_duplicates : bool, default=True
+        If True, the subdivision doesn't include the right boundary of the interval
 
+    Returns
+    -------
+    ndarray
+        Refined 1D array. The length of this array is `n * len(x)`
+        if remove_duplicates and `n + 1 * len(x)` if not.
```

Comment:
```suggestion
        Refined 1D array. The length of this array is `n * (len(x) - 1) + 1`
        if remove_duplicates and `(n + 1) * len(x)` if not.
```

Reply:
Done in 8301506

Reply:
You were too fast! I've just updated the second formula, which was also wrong...

Reply:
Fixed in [957940c](https://github.com/pyccel/psydac/pull/193/commits/957940c2ca7c117d06e90384890b680808990a65)